libraries/ms-helper/inc/thermistor.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+// Thermistor Reading interface for NXRT15XH103FA1B___
+// src:
+// https://www.murata.com/en-global/products/productdetail.aspx?cate=luNTCforTempeSenso&partno=NXRT15XH103FA1B___
+// Requires ADC, GPIO and Interrupts to be initialized.
+#include "adc.h"
+#include "gpio.h"
+#include "interrupt.h"
+#include "soft_timer.h"
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+
+  ADCChannel adc_channel;
+} ThermistorStorage;
+
+typedef struct {
+  uint32_t sibling_resistance;  // milliohms
+
+  ADCChannel adc_channel;
```

Comment:
Why do you require the ADC channel? You should be configuring the GPIO. The idea is for the user of this module to do as little work as possible.

Reply:
We need to know the channel the user is using to get the voltage drop with `adc_read_converted(storage->adc_channel, &reading);` no?

Reply:
The user doesn't care about the channel. They have a GPIO that a thermistor is on. You should be figuring out the rest.